react-intl/index.d.ts

```diff
@@ -14,129 +12,110 @@ declare namespace ReactIntl {
         pluralRuleFunction?: (n: number, ord: boolean) => string;
     }
 
-    function injectIntl<T>(component: React.ComponentClass<InjectedIntlProps & T> | React.StatelessComponent<InjectedIntlProps & T>): React.ComponentClass<T>;
+    type LocaleData = Locale[];
 
-    function addLocaleData(data: Locale[] | Locale): void;
+    interface InjectIntlConfig {
+        intlPropName?: string;
+        withRef?: boolean;
+    }
 
-    function hasLocaleData(localeName: string): boolean;
+    type ComponentConstructor<P> = React.ComponentClass<P> | React.StatelessComponent<P>
+
+    function injectIntl<P>(component: ComponentConstructor<P & InjectedIntlProps>, options?: InjectIntlConfig): React.ComponentClass<P & InjectedIntlProps>;
+
+    function addLocaleData(data: Locale[] | Locale): void;
```

Comment:
According to docs it can be single object or array of objects https://github.com/yahoo/react-intl/wiki/API#addlocaledata

The type itself is not fully correct. I addressed this in my previous PR and fixed the type, but this would be kind of breaking change and as long as tis type is not accessed directly in 99% of cases let's leave it for now.